paasta_tools/contrib/timeouts_metrics_prom.py

```diff
@@ -0,0 +1,59 @@
+#!/opt/venvs/paasta-tools/bin/python
+"""
+The goal is to walk through all smartstack.yaml in yelpsoa directories
+and convert timeout values to the prometheus format.
+
+For every upstream and path, if endpoint timeout is specified, we record
+per endpoint timeout value. If not and if timeout_server_ms is specified,
+we record timeout_server_ms value in path `default`. Otherwise, we record the default
+timeout_server_ms 1000.
+
+# HELP endpoint timeout value defined in yelpsoa-config.
+# TYPE TYPE yelpsoaconfig_endpoint_timeouts_ms gauge
+yelpsoaconfig_endpoint_timeouts_ms{path="/consumer_app/devices/braze/info",upstream="push_notifications.main.egress_cluster"} 10000.0
+yelpsoaconfig_endpoint_timeouts_ms{path="default",upstream="mysql_read_security.main.egress_cluster"} 100.0
+"""
+import os
+
+import yaml
+from prometheus_client import CollectorRegistry
+from prometheus_client import Gauge
+from prometheus_client import write_to_textfile
+
+SOA_DIR = "/nail/etc/services/"
+PROM_OUTPUT_FILE = "/nail/etc/services/.autotune_timeouts.prom"
+
+
+def read_and_write_timeouts_metrics(root, service, prom_metric):
+    with open(os.path.join(root, "smartstack.yaml")) as smartstack_file:
+        smartstack_yaml = yaml.safe_load(smartstack_file)
+    for instance_name, info in smartstack_yaml.items():
+        upstream = service + "." + instance_name + ".egress_cluster"
+        if "endpoint_timeout" in info:
+            for path, endpoint_timeout in info["endpoint_timeout"].items():
+                prom_metric.labels(path, upstream).set(endpoint_timeout)
+        else:
+            if "timeout_server_ms" in info:
```

Comment:
You also don't need this `if` statement: the `info.get` on the line below will return `info["timeout_server_ms"]` if it exists, otherwise it will return 1000

Reply:
fixed that! thanks!